src/gui/editors/AutomationEditor.cpp

```diff
@@ -1308,12 +1308,26 @@ void AutomationEditor::paintEvent(QPaintEvent * pe )
 				}
 
 				float *values = m_pattern->valuesAfter( it.key() );
-				for( int i = 0; i < (it+1).key() - it.key(); i++ )
-				{
 
-					drawLevelTick( p, it.key() + i, values[i],
-									is_selected );
+				float nextValue;
+				if ( m_pattern->valuesAfter( ( it + 1 ).key() ) != NULL )
+					nextValue = *( m_pattern->valuesAfter( ( it + 1 ).key() ) );
+				else
+					nextValue = values[ ( it + 1 ).key() - it.key() -1 ];
+
+				p.setRenderHints( QPainter::Antialiasing, true );
+				QPainterPath path;
+				path.moveTo( QPointF( xCoordOfTick( it.key() ), yCoordOfLevel( 0 ) ) );
+				for( int i = 0; i < ( it + 1 ).key() - it.key(); i++ )
+				{	path.lineTo( QPointF( xCoordOfTick( it.key() + i ), yCoordOfLevel( values[i] ) ) );
+					//drawLevelTick( p, it.key() + i, values[i], is_selected );
+					
 				}
+				path.lineTo( QPointF( xCoordOfTick( ( it + 1 ).key() ), yCoordOfLevel( nextValue ) ) );
+				path.lineTo( QPointF( xCoordOfTick( ( it + 1 ).key() ), yCoordOfLevel( 0 ) ) );
+				path.lineTo( QPointF( xCoordOfTick( it.key() ), yCoordOfLevel( 0 ) ) );
+				p.fillPath( path, graphColor() );
```

Comment:
The `is_selected` variable was used when a particular automation point was selected, but this is disabled and broken atm, it would requre some CSS changes to be properly implemented, and we can save that for when that feature gets fixed.